README.md
# felayout_bluemountain
[![Release](https://img.shields.io/github/release/t3kit/felayout_bluemountain.svg?style=flat-square)](https://github.com/t3kit/felayout_bluemountain/releases)

[![Build Status](https://travis-ci.org/t3kit/felayout_bluemountain.svg?branch=master)](https://travis-ci.org/t3kit/felayout_bluemountain)

Front-End layout for [theme_t3kit_bluemountain](https://github.com/t3kit/theme_t3kit_bluemountain) project

### [CHANGELOG](https://github.com/t3kit/felayout_bluemountain/blob/master/CHANGELOG.md)
### [Contributing to t3kit](https://github.com/t3kit/t3kit/blob/master/CONTRIBUTING.md)

### Required dependencies:

- [Git](https://git-scm.com/)
- [NodeJs](http://nodejs.org/) >=6.0.0
- [NPM](https://github.com/npm/npm) >=3.9.0
- [Bower](http://bower.io/) >=1.7.7 `npm install -g bower`
- [Grunt-cli](http://gruntjs.com/) >=0.1.13 `npm install -g grunt-cli`

### Installation:

First, clone repo:
```bash
git clone git@github.com:t3kit/felayout_bluemountain.git
```

Next, install Bower/NPM dependencies:

```bash
npm install
bower install
```

### Getting Started:

- Run `grunt` to start static server with livereload `localhost:9004`
- Run `grunt +less` to start static server [_same as `grunt`_] plus it generates all Front-End service files plus **LESS** styling for CMS needs, and copy it to `less` folder. _[with livereload]_
- Run `grunt +css` to start static server [_same as `grunt`_] plus it generates all Front-End service files plus **CSS** styling for CMS needs, and copy it to `css` folder. _[with livereload]_
- Run `grunt check` to check HTML/CSS/JS files according project code conventions
- Run `grunt pushSite` to build your static site and push it to separate branch `site`
- Run `grunt pushCss` to compile all Front-End service files plus **CSS** styling for CMS needs, copy it to separate branch `css` and push to remote git server.
- Run `grunt pushLess` to compile all Front-End service files plus **LESS** styling for CMS needs, copy it to separate branch `less` and push to remote git server.


CHANGELOG.md

#### v1.0.0 `January 24, 2017`
- **[FIX]** remove css lint grunt check, fix CI tests ([055ff5f](https://github.com/t3kit/felayout_bluemountain/commit/055ff5f))
- **[BUGFIX]** fix aligning of news pagination and news tags while text is short ([931b94f](https://github.com/t3kit/felayout_bluemountain/commit/931b94f))
- **[FEATURE]** Added 6-7 levels for submenu (Podio feature_28) (#38) ([3a2379a](https://github.com/t3kit/felayout_bluemountain/commit/3a2379a))
- **[FEATURE]** Add third menu-level for main navigation ([836e2b2](https://github.com/t3kit/felayout_bluemountain/commit/836e2b2))
- **[FEATURE]** add new element Simple accordion (Podio feature_34) (#37) ([6d4e32f](https://github.com/t3kit/felayout_bluemountain/commit/6d4e32f))
- **[FEATURE]** add styles for customizing search-text (Podio feature_32) (#36) ([d7e8f75](https://github.com/t3kit/felayout_bluemountain/commit/d7e8f75))
- **[FIX]** add styles for responsive video (Podio bug_60) (#33) ([69bca51](https://github.com/t3kit/felayout_bluemountain/commit/69bca51))
- **[FEATURE]** slider element, add styles for caption alignment (Podio feature_63) (#32) ([a42fc68](https://github.com/t3kit/felayout_bluemountain/commit/a42fc68))
- **[BUGFIX]** fix image scale on ie11, (Podio bug_47) (#31) ([0417793](https://github.com/t3kit/felayout_bluemountain/commit/0417793))
- **[FEATURE]** imageTextLink, add 9 hover effects (animatons) (#29) ([8ec5c09](https://github.com/t3kit/felayout_bluemountain/commit/8ec5c09))
- **[BUGFIX]** Fixed menu looks buggy when scrolling (Podio bug_25) ([0c26b4b](https://github.com/t3kit/felayout_bluemountain/commit/0c26b4b))
- **[FIX]** newsCarousel, fix validation issue (<p> inside of <span> tag)  (#30) ([cea8c94](https://github.com/t3kit/felayout_bluemountain/commit/cea8c94))
- **[FIX]** newsletter subscription error message (Podio bug_31) (#27) ([0e90769](https://github.com/t3kit/felayout_bluemountain/commit/0e90769))
- **[BUGFIX]** no space between pagination and footer on search page (Podio bug_35) (#28) ([2481d07](https://github.com/t3kit/felayout_bluemountain/commit/2481d07))
- **[FEATURE]** add new grid element 'slider container' (Podio Feature_61) (#17) ([c3a942c](https://github.com/t3kit/felayout_bluemountain/commit/c3a942c))
- **[FIX]** fix for making the page not scroll to top when opening the menu (Podio bug_71) ([18b9f29](https://github.com/t3kit/felayout_bluemountain/commit/18b9f29))
- **[FEATURE]** Make the slider autoslide (Podio Feature_41) (#18) ([1fdccea](https://github.com/t3kit/felayout_bluemountain/commit/1fdccea))
- **[TASK]** update favicons ([eafaad6](https://github.com/t3kit/felayout_bluemountain/commit/eafaad6))

***

#### v0.6.0 `October 18, 2016`
- **[STYLE]** fix js formatting ([3ab4879](https://github.com/t3kit/felayout_bluemountain/commit/3ab4879))
- **[FIX]** multiple news carousels (swiper) (Podio bug_42) (#7) ([4685c3f](https://github.com/t3kit/felayout_bluemountain/commit/4685c3f))
- **[FEATURE]** Whole area link (#11) ([9e5d53d](https://github.com/t3kit/felayout_bluemountain/commit/9e5d53d))
- **[FIX]** Print styles (Podio bug_54) (#12) ([b82bd02](https://github.com/t3kit/felayout_bluemountain/commit/b82bd02))
- **[FIX]** wrong class for pagination in solr (Podio bug_45) (#10) ([7981ec2](https://github.com/t3kit/felayout_bluemountain/commit/7981ec2))
- **[FEATURE]** DropdownMenu with columns ([e594088](https://github.com/t3kit/felayout_bluemountain/commit/e594088))
- **[FIX]** use search-button variable also in header top (Podio bug_26) ([4941517](https://github.com/t3kit/felayout_bluemountain/commit/4941517))
- **[FIX]** set the same color in both states of search-button (Podio bug_26) (#5) ([c2e7267](https://github.com/t3kit/felayout_bluemountain/commit/c2e7267))
- **[FIX]** Main navigation disappears after change of screen rotation (Podio bug_33) ([6adccd8](https://github.com/t3kit/felayout_bluemountain/commit/6adccd8))
- **[FEATURE]** Android/iOS detection ([77cf3e2](https://github.com/t3kit/felayout_bluemountain/commit/77cf3e2))

***

#### v0.5.0 `July 18, 2016`
- **[DOC]** add CONTRIBUTING.md file ([e73e384](https://github.com/t3kit/felayout_bluemountain/commit/e73e384))
- **[DOC]** add Travis build status to readme file ([65bcb51](https://github.com/t3kit/felayout_bluemountain/commit/65bcb51))
- **[FIX]** path for js-dotdotdot plugin ([917d330](https://github.com/t3kit/felayout_bluemountain/commit/917d330))
- **[STYLE]** change code formatting ([3a339df](https://github.com/t3kit/felayout_bluemountain/commit/3a339df))
- **[CHORE]** disable force option in jscs and jshint ([0b8efae](https://github.com/t3kit/felayout_bluemountain/commit/0b8efae))
- **[UPDATE]** Bower dependencies ([fba12e9](https://github.com/t3kit/felayout_bluemountain/commit/fba12e9))
- **[TEST]** add Travis CI integration ([c7da81b](https://github.com/t3kit/felayout_bluemountain/commit/c7da81b))
- **[UPDATE]** package.json devDependencies ([374cd55](https://github.com/t3kit/felayout_bluemountain/commit/374cd55))
- **[UPDATE]** update node.js (>=6.0.0) and npm (>=3.9.0) dependencies ([a3079fd](https://github.com/t3kit/felayout_bluemountain/commit/a3079fd))

***

#### v0.4.0 `July 8, 2016`

dev/js/main/header/header.js
// =================================
// Global variables (jshint):

/*global touchSupport*/
/*global isAndroid*/
///*global isIOS*/
// =================================

jQuery(function($) {
    // Caching variables
    var $html = $('html');
    var $mainNavigation = $('.js__main-navigation');
    var $openSubMenuLink = $('.js__main-navigation__open-sub-menu-link');
    var $mainNavigationItemsList = $mainNavigation.find('.js__main-navigation__items-list').children('li');

    //var $mainNavigationItemsListSub = ('.main-navigation__item._sub');
    var $dropdownMenuWithColumns = $('.js__dropdown-menu-with-columns .js__main-navigation__item._sub');
    if (!touchSupport) {
        $dropdownMenuWithColumns.hover(function() {
            $(this).toggleClass('open');
        });
    }

    // Cleanup function to clean unneeded classes
    var cleanup = function() {
        $mainNavigationItemsList.removeClass('_open-mobile-dropdown _open-tablet-dropdown');
        $html.removeClass('mobile-menu-opened');

        if (window.matchMedia('(min-width: 992px)').matches) {
            $('.js__navigation__items-wrp').show();
        } else {
            $('.js__navigation__items-wrp').hide();
        }
    };

    // Add click event to dropdown link on mobile devices.
    $openSubMenuLink.on('click', function(e) {
        e.preventDefault();

        if (window.matchMedia('(min-width: 992px)').matches) {
            $mainNavigationItemsList.not($(this).parents()).removeClass('_open-tablet-dropdown');
            $(this).parents('.main-navigation__item').toggleClass('_open-tablet-dropdown');
        } else {
            $(this).parents('.main-navigation__item').toggleClass('_open-mobile-dropdown');
        }
    });

    // detect if we cross 992px window width.
    window.matchMedia('(min-width: 992px)').addListener(cleanup);

    var mobileMenuAnimationComplete = true;
    $('.js__main-navigation__toggle-btn').on('click', function(e) {
        e.preventDefault();
        if (mobileMenuAnimationComplete) {
            mobileMenuAnimationComplete = false;
            $html.toggleClass('mobile-menu-opened');
        }
        $('.js__navigation__items-wrp').not(':animated').slideToggle(300, function() {
            mobileMenuAnimationComplete = true;
        });
    });

});

// ====== class fo fixed main navigation bar   =======
jQuery(function($) {
    var navbar = $('.js__main-navigation');

    if (navbar.length) {
        var offsetTop = navbar.offset().top;

        // function that calculates offsetTop-value.
        var calcOffsetTop = function() {
            if (window.matchMedia('(min-width: 992px)').matches) {
                var navbarPos = navbar.css('position');
                offsetTop = $('header').height() - (navbarPos === 'fixed' ? 0 : navbar.outerHeight());
            }
        };

        // detect if we cross 992px window width.
        window.matchMedia('(min-width: 992px)').addListener(calcOffsetTop);

        $(window).on('load scroll', function() {
            var scrollPos = $(window).scrollTop();
            if (scrollPos > offsetTop) {
                $('body').addClass('main-navigation-fixed');
            } else {
                $('body').removeClass('main-navigation-fixed');
            }
        });
    }
});

jQuery(function($) {
    var $mainNavigationSearchBtn = $('.js__main-navigation__search-btn');
    var $mainNavigationSearchBox = $('.js__main-navigation__search-box');
    var $mainNavigationSearchBoxOverlay = $('.js__main-navigation__search-box-overlay');

    var $languageMenuOverlay = $('.js__header-top__language-menu-overlay');
    var $languageMenuBtn = $('.js__header-top__language-menu-btn');
    var $languageMenuBox = $('.js__header-top__language-menu-box');
    var $languageMenuBoxCloseBtn = $('.js__header-top__language-menu-box-close-btn');

    $mainNavigationSearchBtn.on('click', function(e) {
        e.preventDefault();
        $(this).toggleClass('_search-close-btn');
        $mainNavigationSearchBox.toggleClass('_search-box-visible');
        if ($mainNavigationSearchBox.hasClass('_search-box-visible')) {
            $mainNavigationSearchBox.find('input[type="search"]').focus();
        }
        $mainNavigationSearchBoxOverlay.toggleClass('_search-box-overlay-visible');
    });
    $mainNavigationSearchBoxOverlay.on('click', function() {
        $(this).toggleClass('_search-box-overlay-visible');
        $mainNavigationSearchBtn.toggleClass('_search-close-btn');
        $mainNavigationSearchBox.toggleClass('_search-box-visible');
    });

    $languageMenuBtn.on('click', function(e) {
        e.preventDefault();
        $languageMenuBox.addClass('_language-menu-box-visible');
        $languageMenuOverlay.toggleClass('_language-menu-box-overlay-visible');
    });
    $languageMenuOverlay.on('click', function() {
        $(this).toggleClass('_language-menu-box-overlay-visible');
        $languageMenuBox.removeClass('_language-menu-box-visible');
    });
    $languageMenuBoxCloseBtn.on('click', function() {
        $languageMenuOverlay.toggleClass('_language-menu-box-overlay-visible');
        $languageMenuBox.removeClass('_language-menu-box-visible');
    });

});


dev/js/main/plugins/news/newsCarousel.js
(function($) {
    'use strict';

    // document load event
    $(document).ready(function() {

        // initialize swiper when document ready
        // http://idangero.us/swiper/api/
        $('.js__news-carousel').each(function(){
          $(this).swiper({
            nextButton: $(this).parent().find('.js__news-carousel__btn-next'),
            prevButton: $(this).parent().find('.js__news-carousel__btn-prev'),
            pagination: '.js__news-carousel__pagination',
            paginationClickable: true,
            slidesPerView: 4,
            preloadImages: false,
            spaceBetween: 30,

            // Responsive breakpoints
            breakpoints: {

                // when window width is <= 480px
                500: {
                    slidesPerView: 1
                },
                // when window width is <= 768px
                767: {
                    slidesPerView: 2
                },
                // when window width is <= 992px
                991: {
                    slidesPerView: 3
                },
                // when window width is <= 1199px
                1199: {
                    slidesPerView: 4
                }
            }
          });
        });
    });

})(jQuery);


main.js

// header
// =================================
// Global variables (jshint):

/*global touchSupport*/
// =================================

jQuery(function($) {
    // Caching variables
    var $html = $('html');
    var $mainNavigation = $('.js__main-navigation');
    var $openSubMenuLink = $('.js__main-navigation__open-sub-menu-link');
    var $mainNavigationItemsList = $mainNavigation.find('.js__main-navigation__items-list').children('li');
    var $secondNavLevel = $('.second-navigation-level');
    var $thirdNavLevel = $('.third-navigation-level');
    var $openThirdMenuLink = $('.js__main-navigation__open-third-menu-link');
    var $dropdownMenuWithColumns = $('.js__dropdown-menu-with-columns .js__main-navigation__item._sub');

    if (!touchSupport) {
        $dropdownMenuWithColumns.hover(function() {
            $(this).toggleClass('open');
        });
    }

    // Set class for third-navigation-level to handle position on left or right
    var setThirdMenuPosition = function() {
        if ($thirdNavLevel.length) {
            if (window.matchMedia('(min-width: 992px)').matches) {
                $secondNavLevel.each(function() {
                    var offsetRight = $(window).width() - $(this).offset().left - $(this).outerWidth();
                    var secondNavLevelWidth = $(this).width();
                    var thirdNavLevelWidth = $(this).find($thirdNavLevel).width();
                    if (offsetRight < thirdNavLevelWidth) {
                        $(this).find($thirdNavLevel).css('left', -secondNavLevelWidth);
                    } else {
                        $(this).find($thirdNavLevel).css('left', secondNavLevelWidth);
                    }
                });
            } else {
                $thirdNavLevel.css('left', 'auto');
            }
        }
    };
    // Initial call for function
    setThirdMenuPosition();
    // Cleanup function to clean unneeded classes
    var cleanup = function() {
        $mainNavigation.find('.js__main-navigation__items-list').find('li').removeClass('_open-mobile-dropdown _open-tablet-dropdown');
        $html.removeClass('mobile-menu-opened');

        if (window.matchMedia('(min-width: 992px)').matches) {
            $('.js__navigation__items-wrp').show();
        } else {
            $('.js__navigation__items-wrp').hide();
        }

        //Set timeout for third menu position to load the width
        window.setTimeout(function() {
            setThirdMenuPosition();
        }, 500);
    };

    // Add click event to dropdown link on mobile devices.
    $openSubMenuLink.on('click', function(e) {
        e.preventDefault();
        if (window.matchMedia('(min-width: 992px)').matches) {
            $mainNavigationItemsList.not($(this).parents()).removeClass('_open-tablet-dropdown');
            $(this).parents('.main-navigation__item').toggleClass('_open-tablet-dropdown');
        } else {
            $(this).parents('.main-navigation__item').toggleClass('_open-mobile-dropdown');
        }
    });

    // Add click event to second menu dropdown link on mobile devices.
    $openThirdMenuLink.on('click', function(e) {
        e.preventDefault();
        if (window.matchMedia('(min-width: 992px)').matches) {
            $('.main-navigation__sub-item').not($(this).parents('.main-navigation__sub-item')).removeClass('_open-tablet-dropdown');
            $(this).parents('.main-navigation__sub-item').toggleClass('_open-tablet-dropdown');
        } else {
            $(this).parents('.main-navigation__sub-item').toggleClass('_open-mobile-dropdown');
        }
    });

    var mobileMenuAnimationComplete = true;
    $('.js__main-navigation__toggle-btn').on('click', function(e) {
        e.preventDefault();
        if (mobileMenuAnimationComplete) {
            mobileMenuAnimationComplete = false;
            $html.toggleClass('mobile-menu-opened');
        }
        $('.js__navigation__items-wrp').not(':animated').slideToggle(300, function() {
            mobileMenuAnimationComplete = true;
        });
    });

    // detect if we cross 992px window width.
    window.matchMedia('(min-width: 992px)').addListener(cleanup);
});

// ====== class fo fixed main navigation bar   =======
jQuery(function($) {
    var navbar = $('.js__main-navigation');

    if (navbar.length) {
        var offsetTop = navbar.offset().top;

        // function that calculates offsetTop-value.
        var calcOffsetTop = function() {
            if (window.matchMedia('(min-width: 992px)').matches) {
                var navbarPos = navbar.css('position');
                offsetTop = $('header').height() - (navbarPos === 'fixed' ? 0 : navbar.outerHeight());
            }
        };

        // detect if we cross 992px window width.
        window.matchMedia('(min-width: 992px)').addListener(calcOffsetTop);

        $(window).on('load scroll', function() {
            var scrollPos = $(window).scrollTop();
            if (scrollPos > offsetTop) {
                $('body').addClass('main-navigation-fixed');
            } else {
                $('body').removeClass('main-navigation-fixed');
            }
        });
    }
});

jQuery(function($) {
    var $mainNavigationSearchBtn = $('.js__main-navigation__search-btn');
    var $mainNavigationSearchBox = $('.js__main-navigation__search-box');
    var $mainNavigationSearchBoxOverlay = $('.js__main-navigation__search-box-overlay');

    var $languageMenuOverlay = $('.js__header-top__language-menu-overlay');
    var $languageMenuBtn = $('.js__header-top__language-menu-btn');
    var $languageMenuBox = $('.js__header-top__language-menu-box');
    var $languageMenuBoxCloseBtn = $('.js__header-top__language-menu-box-close-btn');

    $mainNavigationSearchBtn.on('click', function(e) {
        e.preventDefault();
        $(this).toggleClass('_search-close-btn');
        $mainNavigationSearchBox.toggleClass('_search-box-visible');
        if ($mainNavigationSearchBox.hasClass('_search-box-visible')) {
            $mainNavigationSearchBox.find('input[type="search"]').focus();
        }
        $mainNavigationSearchBoxOverlay.toggleClass('_search-box-overlay-visible');
    });
    $mainNavigationSearchBoxOverlay.on('click', function() {
        $(this).toggleClass('_search-box-overlay-visible');
        $mainNavigationSearchBtn.toggleClass('_search-close-btn');
        $mainNavigationSearchBox.toggleClass('_search-box-visible');
    });

    $languageMenuBtn.on('click', function(e) {
        e.preventDefault();
        $languageMenuBox.addClass('_language-menu-box-visible');
        $languageMenuOverlay.toggleClass('_language-menu-box-overlay-visible');
    });
    $languageMenuOverlay.on('click', function() {
        $(this).toggleClass('_language-menu-box-overlay-visible');
        $languageMenuBox.removeClass('_language-menu-box-visible');
    });
    $languageMenuBoxCloseBtn.on('click', function() {
        $languageMenuOverlay.toggleClass('_language-menu-box-overlay-visible');
        $languageMenuBox.removeClass('_language-menu-box-visible');
    });

});


// elements
/* global Swiper*/
(function($) {
    'use strict';

    // document load event
    $(document).ready(function() {

        // initialize swiper when document ready
        // http://idangero.us/swiper/api/
        var $elem = $('.js__img-slider');
        $elem.each(function() {
            var time = $(this).attr('data-autoplay');
            var slider = new Swiper($(this), {
                nextButton: '.js__img-slider__btn-next',
                prevButton: '.js__img-slider__btn-prev',
                pagination: '.js__img-slider__pagination',
                paginationClickable: true,
                preloadImages: false,
                lazyLoading: true,
                watchSlidesVisibility: true,
                lazyLoadingInPrevNext: true,
                speed: 600,
                autoplay: time
            });
            // Makes it possible to skip between slider images if they have links, using the tab button
            slider.container.on('focus', 'a', function(e) {
                //Index of focused slide
                var focusIndex = $(e.target).parents('.swiper-slide').index();
                //Reset scrollLeft set by browser on focus
                slider.container.scrollLeft(0);

                // IE fix
                setTimeout(function() {
                    slider.container.scrollLeft(0);
                }, 0);

                //Slide to focused slide
                slider.slideTo(focusIndex);
            });
        });
    });
})(jQuery);

(function($) {
    'use strict';

    // document load event
    $(document).ready(function() {

        // initialize swiper when document ready
        // http://idangero.us/swiper/api/
        $('.js__logo-carousel').swiper({
            nextButton: '.js__logo-carousel__btn-next',
            prevButton: '.js__logo-carousel__btn-prev',
            slidesPerView: 5,
            preloadImages: false,
            lazyLoading: true,
            watchSlidesVisibility: true,
            lazyLoadingInPrevNext: true,
            spaceBetween: 20,
            autoplay: 2500,

            // Responsive breakpoints
            breakpoints: {

                // when window width is <= 480px
                480: {
                    slidesPerView: 1
                },

                // when window width is <= 600px
                600: {
                    slidesPerView: 2
                },

                // when window width is <= 768px
                768: {
                    slidesPerView: 3
                },

                // when window width is <= 992px
                992: {
                    slidesPerView: 4
                }
            }
        });
    });

})(jQuery);

(function($) {
    'use strict';

    // document load event
    $(document).ready(function() {
        // Parallax
        // https://github.com/nk-o/jarallax
        if (!$('html').hasClass('IE')) { // disabled in IE since scrolling looks jerky
            $('.parallax-img').jarallax({
                type: 'scroll', //scroll, scale, opacity, scroll-opacity, scale-opacit
                speed: 0.5,
                noAndroid: false,
                noIos: true
            });
            $('.parallax-video').jarallax({
                type: 'scroll', //scroll, scale, opacity, scroll-opacity, scale-opacit
                speed: 0.5,
                noAndroid: true,
                noIos: true
            });
        }
    });

})(jQuery);

/* global Swiper*/
(function($) {
    'use strict';
    // document load event
    $(document).ready(function() {
        var $swiperContainerWrapper = $('.js__slider-container__wrapper');
        $swiperContainerWrapper.each(function() {
            $(this).children().wrap('<div class="swiper-slide slider-container__slide js__slider-container__slide"></div>');
        });

        var $swiperContainer = $('.js__slider-container__container');
        $swiperContainer.each(function() {
            var time = $(this).attr('data-autoplay');
            var loopParam = $(this).attr('data-loop');
            var amountOfSlides = parseInt($(this).attr('data-slidesperview'));
            var effectName = $(this).attr('data-effect');
            var transition = $(this).attr('data-speed');
            var widthForMobile;
            var widthForTablet;
            var widthForLaptop;
            var widthForMediumLaptop;
            if (amountOfSlides >= 4) {
                widthForMobile = 1,
                widthForTablet = 2,
                widthForLaptop = 3,
                widthForMediumLaptop = 4;
            } else if (amountOfSlides === 2) {
                widthForMobile = 1,
                widthForTablet = 1,
                widthForLaptop = 1,
                widthForMediumLaptop = 2;
            } else if (amountOfSlides === 1) {
                widthForMobile = 1,
                widthForTablet = 1,
                widthForLaptop = 1,
                widthForMediumLaptop = 1;
            } else {
                widthForMobile = 1,
                widthForTablet = 2,
                widthForLaptop = 2,
                widthForMediumLaptop = 3;
            }
            var slider = new Swiper($(this), {
                containerModifierClass:'js__slider-container__container',
                wrapperClass:'js__slider-container__wrapper',
                slideClass:'js__slider-container__slide',
                nextButton: $(this).parent().find('.js__slider-container__btn-next'),
                prevButton: $(this).parent().find('.js__slider-container__btn-prev'),
                pagination: $(this).parent().find('.js__slider-container__pagination'),
                paginationClickable: true,
                speed: parseInt(transition),
                loop:loopParam,
                autoplay: time,
                effect:effectName,
                watchSlidesVisibility: true,
                spaceBetween: 20,
                preloadImages: false,
                lazyLoading: true,
                lazyLoadingInPrevNext: true,
                slidesPerView: amountOfSlides,
                breakpoints:{
                    // Responsive breakpoints
                    480: {
                        slidesPerView:widthForMobile
                    },
                    767: {
                        slidesPerView:widthForTablet
                    },
                    992: {
                        slidesPerView:widthForLaptop
                    },
                    1024:{
                        slidesPerView:widthForMediumLaptop
                    }
                },
                coverflow: {
                    rotate: 90,
                    stretch: 0,
                    depth: 200,
                    modifier: 1,
                    slideShadows: false
                },
                cube: {
                    slideShadows: false,
                    shadow: false
                },
                fade: {
                    crossFade: true
                },
                flip: {
                    slideShadows: false,
                    limitRotation: true
                }
            });
            // Makes it possible to skip between slider images if they have links, using the tab button
            slider.container.on('focus', 'a', function(e) {
                //Index of focused slide
                var focusIndex = $(e.target).parents('.slider-container__slide').index();
                //Reset scrollLeft set by browser on focus
                slider.container.scrollLeft(0);
                // IE fix
                setTimeout(function() {
                    slider.container.scrollLeft(0);
                }, 0);
                //Slide to focused slide
                slider.slideTo(focusIndex);
            });
        });
    });
})(jQuery);

(function($) {
    'use strict';

    // document load event
    $(document).ready(function() {

        var $paragraph = $('.js__img-text-link');

        $paragraph.dotdotdot({
            height: 60
        });

    });

})(jQuery);


// plugins
(function($) {
    'use strict';

    // document load event
    $(document).ready(function() {

        // initialize swiper when document ready
        // http://idangero.us/swiper/api/
        $('.js__news-carousel').each(function() {
            $(this).swiper({
                nextButton: $(this).parent().find('.js__news-carousel__btn-next'),
                prevButton: $(this).parent().find('.js__news-carousel__btn-prev'),
                pagination: '.js__news-carousel__pagination',
                paginationClickable: true,
                slidesPerView: 4,
                preloadImages: false,
                spaceBetween: 30,

                // Responsive breakpoints
                breakpoints: {

                    // when window width is <= 480px
                    500: {
                        slidesPerView: 1
                    },
                    // when window width is <= 768px
                    767: {
                        slidesPerView: 2
                    },
                    // when window width is <= 992px
                    991: {
                        slidesPerView: 3
                    },
                    // when window width is <= 1199px
                    1199: {
                        slidesPerView: 4
                    }
                }
            });
        });
    });

})(jQuery);

// news Timeline
$('.js__news-timeline__item').on('click', function(e) {
    if ($(this).hasClass('collapsed')){
        e.preventDefault();
        $(this).removeClass('collapsed');
        $(this).closest('.js__news-timeline__item-wrp').find('.js__news-timeline__date').addClass('open');
    }
});

// news Cards
$('.js__news-cards__dotdotdot').dotdotdot({
    watch: 'window'
});

// news Simple list
$('.js__news-simple-list__dotdotdot').dotdotdot({
    watch: 'window'
});


/*global tx_solr_suggestUrl*/
/*global Awesomplete*/
/*global touchSupport*/

// global var
// container for Search suggestion data
var mainSearchInputList = {};

(function($) {
    'use strict';

    // init function
    var searchSuggestFn = function() {

        // ============================
        // Search Suggest DATA-API
        // ============================
        $('[data-search="searchSuggest"]').each(function(index, el) {
            mainSearchInputList['searchItem' + index] = new Awesomplete(el, {
                list: [],
                maxItems: 20,
                minChars: 2,
                autoFirst: true
            });
            var req = false;
            $(this).on('keyup.search.suggest', function(e) {
                var c = e.keyCode;
                if (c === 13 ||c === 27 || c === 38 || c === 40) {
                    return;
                }
                var that = $(this);
                var fetchSuggestData = function() {
                    if (!req) {
                        req = true;
                        $.ajax({
                            url: tx_solr_suggestUrl, // jscs:ignore requireCamelCaseOrUpperCaseIdentifiers
                            dataType: 'json',
                            data:{
                                termLowercase: that.val().toLowerCase(),
                                termOriginal: that.val(),
                                L: jQuery('div.tx-solr input[name="L"]').val()
                            },
                            success: function(data) {
                                req = false;
                                var arrr = [];
                                $.each(data, function(term) {
                                    arrr.push(term);
                                });
                                mainSearchInputList['searchItem' + index]._list = arrr;
                                mainSearchInputList['searchItem' + index].evaluate();
                            }
                        });
                    }
                };
                typeof tx_solr_suggestUrl !== 'undefined' && fetchSuggestData(); // jscs:ignore requireCamelCaseOrUpperCaseIdentifiers
            });
            $(this).on('awesomplete-selectcomplete', function() {
                $(this).closest('form').submit();
            });
        });
        // =============end==============

    };

    // document load event
    $(document).ready(function() {
        if (!touchSupport && $(window).width() >= 992) {
            searchSuggestFn();
        }
    });

})(jQuery);


// ########## general.js ###########

(function($) {
    'use strict';

    // document load event
    $(document).ready(function() {

        // Image Lightbox
        // initialize simpleLightbox when document ready
        // https://github.com/andreknieriem/simplelightbox
        $('.lightbox').simpleLightbox({
            captionType: 'data',
            captionsData: 'caption',
            captionPosition: 'outside',
            heightRatio: 0.6
        });
    });

})(jQuery);

// ^^^^^^^^^^ general.js ^^^^^^^^^^^

